#! /bin/sh /usr/share/dpatch/dpatch-run
## rrdtool_uninitialized_fix.dpatch by Florian Forster <octo@verplant.org>
##
## DP: rrdtool plugin: Make absolutely sure two local variables are
## DP: initialized.

@DPATCH@

diff a/src/rrdtool.c b/src/rrdtool.c
--- a/src/rrdtool.c
+++ b/src/rrdtool.c
@@ -624,6 +624,9 @@ static void *rrd_queue_thread (void *data)
 		int    status;
 		int    i;
 
+		values = NULL;
+		values_num = 0;
+
                 pthread_mutex_lock (&queue_lock);
                 /* Wait for values to arrive */
                 while (true)
